### PR TITLE
Xcode 10 & more

### PR DESCRIPTION
**Pull request by shagedorn**

### CODE REVIEW

File: ISHPermissionKit/Requests/ISHPermissionRequestUserNotification.m [hunk 1]
```diff
@@ -65,6 +59,7 @@ - (ISHPermissionState)permissionStateForNotificationSettings:(UNNotificationSett
         case UNAuthorizationStatusDenied:
             return ISHPermissionStateDenied;
 
+        case UNAuthorizationStatusProvisional:
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequestUserNotification.m]: This is important:

You can ask for provisional permission (you must do this explicitly via `desiredOptions`, which will allow you to send one quite (no alert/sound, but in the notification center) notification. Users can change their settings directly from notification center.

After requesting provisional permission, the auth state stays in `UNAuthorizationStatusProvisional` until the user has made a decision. During that time, you may also request general permission, which is why we treat it as "NotDetermined".

Comment (mollidor) [reply ISHPermissionKit/Requests/ISHPermissionRequestUserNotification.m]: thanks for the detailed explanation. This makes for a great comment in the code I reckon. Should be valuable later on when we come across this and 🤔.

**Comment by shagedorn** [comment]
@mollidor decided to bump versions here since I already touched the podspec. Slightly modified the short version strings:

versions

**Comment by mollidor** [comment]
Not sure modifying the short version this way is a great idea.
According to [Apple's doc](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-111349)

> The release version number is a string composed of three period-separated integers.

I doubt it will cause issues as this is Debug and Coverage only, but if it bites it will be tricky to debug.

**Comment by shagedorn** [comment]
@mollidor Thx, wasn't aware of it. Will remove the suffix.

**Comment by mollidor** [comment]
It seems all of this is just convention though and there is no strict rules as the same docs say later for [CFBundleVersion](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364)

> While developing a new version of your app, you can include a suffix after the number that is being updated; for example 3.1.3a1. The character in the suffix represents the stage of development for the new version. For example, you can represent development, alpha, beta, and final candidate, by d, a, b, and fc. The final number in the suffix is the build version, which cannot be 0 and cannot exceed 255. When you release the new version of your app, remove the suffix.

**Comment by mollidor** [comment]
... and the title of the PR should actually read "Xcode 10 $ less" 🙃

**Pull request merged**